rtps-r52/shell-tests.c

```diff
@@ -204,3 +204,32 @@ rtems_shell_cmd_t shell_cmd_test_link_shmem_trch = {
     NULL, NULL,                                /* alias, next */
     0, 0, 0                                    /* mode, uid, gid */
 };
+
+static int shell_test_mbox_rtps_hpps(int argc RTEMS_UNUSED,
+                                         char *argv[] RTEMS_UNUSED)
+{
+    if (argc <= 1) {
+        fprintf(stderr, "ERROR: TEST REQUIRES 2 MBOXES\n");
+        return -1;
+    }
+
+    int mbox_in = atoi(argv[1]);
+    int mbox_out = atoi(argv[2]);
+
+    if (mbox_in > 31 || mbox_in < 0 || mbox_out > 31 || mbox_out < 0) {
+        fprintf(stderr, "ERROR: INVALID MBOX NUMBERS\n");
+        return -1;
+    }
```

Comment:
In the rest of the code we declare mailbox instances to be of `unsigned` type.  Let's be consistent here.  It also simplifies your parameter checking -- you'll only need to check that the value is not >=32.

In fact, I think that the index checking should be performed further down the function call chain in `link-mbox.c: link_mbox_connect`, since it wraps the driver and can make use of its header macros.  So for example, at the top of that function you can just check that `idx_from` and `idx_to` are `< HPSC_MBOX_CHANNELS`, otherwise return an error there.  Then everything that uses mailbox links gets the benefit of the range check and the callers don't have to have explicit knowledge about the total number of mailboxes.  The only reason this check was never added there was because there's an `assert` in the driver and all the channel indices that are passed through mailbox-related functions have thus far been known compile-time.